optihood/energy_network.py

```diff
@@ -1,582 +1,648 @@
+        data = pd.ExcelFile(filePath)
+        nodesData = self.createNodesData(data, filePath, numberOfBuildings)
+        nodesData["buses"]["excess costs"] = nodesData["buses"]["excess costs indiv"]
+        nodesData["electricity_cost"]["cost"] = nodesData["electricity_cost"]["cost indiv"]
+
+        demandProfiles = {}
+        if clusterSize:
+            for i in range(1, numberOfBuildings + 1):
+                demandProfiles[i] = pd.concat(
+                    [nodesData["demandProfiles"][i].loc[d] for d in clusterSize.keys()])
+            electricityImpact = pd.concat(
+                [nodesData["electricity_impact"].loc[d] for d in clusterSize.keys()])
+            electricityCost = pd.concat(
+                [nodesData["electricity_cost"].loc[d] for d in clusterSize.keys()])
+            weatherData = pd.concat([nodesData['weather_data'][
+                                         nodesData['weather_data']['time.mm'] == int(d.split('-')[1])][
+                                         nodesData['weather_data']['time.dd'] == int(d.split('-')[2])][
+                                         ['gls', 'str.diffus', 'tre200h0','ground_temp']] for d in clusterSize.keys()])
+
+            nodesData["demandProfiles"] = demandProfiles
+            nodesData["electricity_impact"] = electricityImpact
+            nodesData["electricity_cost"] = electricityCost
+            nodesData["weather_data"] = weatherData
+
+        self._convertNodes(nodesData, opt)
+        logging.info("Nodes from Excel file {} successfully converted".format(filePath))
+        self.add(*self._nodesList)
+        logging.info("Nodes successfully added to the energy network")
+
+    def createNodesData(self, data, filePath, numBuildings):
+        self.__noOfBuildings = numBuildings
+        nodesData = {
+            "buses": data.parse("buses"),
+            "grid_connection": data.parse("grid_connection"),
+            "commodity_sources": data.parse("commodity_sources"),
+            "solar": data.parse("solar"),
+            "transformers": data.parse("transformers"),
+            "demand": data.parse("demand"),
+            "storages": data.parse("storages"),
+            "stratified_storage": data.parse("stratified_storage"),
+            "csv_paths": data.parse("csv_data")
+        }
+        # update stratified_storage index
+        nodesData["stratified_storage"].set_index("label", inplace=True)
+
+        # extract input data from CSVs
+        electricityImpactPath = nodesData["csv_paths"].loc[nodesData["csv_paths"]["name"] == "electricity_impact", "path"].iloc[0]
+        electricityCostPath = nodesData["csv_paths"].loc[nodesData["csv_paths"]["name"] == "electricity_cost", "path"].iloc[0]
+        demandProfilesPath = nodesData["csv_paths"].loc[nodesData["csv_paths"]["name"] == "demand_profiles", "path"].iloc[0]
+        weatherDataPath = nodesData["csv_paths"].loc[nodesData["csv_paths"]["name"] == "weather_data", "path"].iloc[0]

```

Comment:
'csv_paths' should be corrected to 'profiles' as per the previous comment (look at the latest version of the file)

Reply:
We decided to change location of the file for energy cost in the sheet 'commodity sources' in place of fix cost